mesh/mesh.go

```diff
@@ -499,21 +500,14 @@ func (msh *Mesh) reInsertTxsToPool(applied *types.Block, notApplied []*types.Blo
 	if len(missing) > 0 {
 		msh.With().Error("could not reinsert transactions", log.Int("missing", len(missing)), l)
 	}
-
-	grouped, accounts, err := msh.removeFromUnappliedTxs(returnedTxs)
-	if err != nil {
+	if err := msh.deleteTransactions(returnedTxs...); err != nil {
```

Comment:
this doesn't means delete the entry from DB. this means set the TX to applied. we should never delete TXs from db.

Reply:
> this doesn't means delete the entry from DB. this means set the TX to applied. 

i think this is false. applied is set for transactions that were *applied* to the state.

> we should never delete TXs from db.

i think we should. if tx is not applied we are adding it back to mempool. later it will be included with another block or dropped from the mempool

Reply:
right. this is different from current behavior but i agree to delete here.

Reply:
i misunderstood current behavior. we were removing from unnaplied db but not from transactions db. maybe i should keep them and rename applied to something that makes more sense

Reply:
i think maybe the node can enter some unexpected state if a block wasn't applied and later tortoise changed opinion and block became applied. so i guess it is safer to preserve original behavior for now

Reply:
yes. a TX is
added to db when a proposal is received, or when a block is synced

> i think maybe the node can enter some unexpected state if a block wasn't applied and later tortoise changed opinion and block became applied. so i guess it is safer to preserve original behavior for now

yes. a TX is added to db only when a proposal is received via gossip or when a block is fetched. 
if you delete here, you will have to add it back from the block when it become contextually valid from invalid. that will make sense too.

Reply:
i also thought about that. the reason why i wasn't sure that this is safe is because nodes during self-healing will vote for a block even if transactions are not stored locally. this may lead to a condition where the block becomes valid during self-healing, but data is not available to honest nodes.

so until we implement voting against the blocks with missing data we probably can't do what i wanted